iothub/device/src/Transport/ClientWebSocketTransport.cs

```diff
@@ -63,13 +63,14 @@ public override bool WriteAsync(TransportAsyncCallbackArgs args)
             Fx.AssertAndThrow(args.CompletedCallback != null, "must have a valid callback");
             args.Exception = null; // null out any exceptions
 
+            // We can't await this properly because the abstract method, defined by the AMQP layer, is not Task<bool>.
             Task taskResult = WriteImplAsync(args);
             if (WriteTaskDone(taskResult, args))
```

Comment:
This code looks fundamentally broken to me. I really didn't touch it, but I have big concerns.